src/embedded_flash/store/mod.rs

```diff
@@ -449,6 +463,31 @@ impl<S: Storage, C: StoreConfig> Store<S, C> {
     /// The provided index must point to the beginning of an entry.
     fn delete_index(&mut self, index: Index) {
         self.update_word(index, |format, word| format.set_deleted(word));
+        self.wipe_sensitive_data(index);
+    }
+
+    /// Wipes the data of a sensitive entry.
+    ///
+    /// If the entry at the provided index is sensitive, overwrites the data with zeroes. Otherwise,
+    /// does nothing.
+    fn wipe_sensitive_data(&mut self, mut index: Index) {
+        let entry = self.read_entry(index);
+        debug_assert!(self.format.is_present(entry));
+        debug_assert!(self.format.is_deleted(entry));
+        if self.format.is_internal(entry) || !self.format.is_sensitive(entry) {
+            // No need to wipe the data.
+            return;
+        }
+        let gap = self.format.entry_gap(entry);
+        let data = &entry[gap.start..][..gap.length];
```

Comment:
Since the next line iterates on the bytes anyway, I think it'd be more readable as `.iter().skip(gap.start).take(gap.length)`. I think that the "double slicing" `entry[gap.start..][..gap.length]` is more confusing.

Reply:
I created a helper function for that in `ByteGap` since it's used twice. And I used `[start .. start + length]` which is fine because `start` is a variable. I find it more readable than chaining iterator transformations.